Sources/AWSSDKSwiftCore/AWSClient.swift

```diff
@@ -15,23 +15,42 @@
 import AsyncHTTPClient
 import AWSSignerV4
 import AWSXML
+import Baggage
 import Dispatch
 import struct Foundation.Data
 import class Foundation.JSONDecoder
 import class Foundation.JSONSerialization
 import struct Foundation.URL
 import struct Foundation.URLQueryItem
+import Instrumentation
 import Logging
 import Metrics
 import NIO
 import NIOConcurrencyHelpers
 import NIOHTTP1
 import NIOTransportServices
+import TracingInstrumentation
+
+// TODO: instrument initialization
+// TODO: create internal spans for signing, parsing?
+
+// TODO: remove and use optional parameter or dont provide default value
+public struct AWSClientContext: AWSClient.Context {
```

Comment:
@adam-fowler what is your decision here? should the context be optional? defaults to (static) "empty" value?